pkg/datapath/loader/loader.go

```diff
@@ -346,6 +346,13 @@ func removeObsoleteNetdevPrograms(devices []string) error {
 			continue
 		}
 
+		// Remove the per-device bpffs directory containing pinned links and
+		// per-endpoint maps.
+		bpffsPath := bpffsDeviceDir(bpf.CiliumPath(), l)
+		if err := bpf.Remove(bpffsPath); err != nil {
+			log.WithError(err).WithField(logfields.Device, l.Attrs().Name)
+		}
+
```

Comment:
Is this chunk a bugfix? Seems we should have always removed this dir or is this the first link there?

Reply:
No, given this is the first time we're using tcx links.